### PR TITLE
Persistent private key as a secret generates tokens on each request

### PR DESCRIPTION
**Pull request by Alex-Monahan**
Let me know if you have feedback! There are 2 completely identical pieces of code in read and copy, but I don't know enough about passing pointers around to consolidate. Happy to clean that up if needed though! I just reached my C++ limit for the day. 

Basically, if you copy the private key out of the JSON file and use it as your secret, you never have to reauthenticate!

I have not tested with a persistent DuckDB secret, but I don't think this would conflict with that. 

The httplib file was copied out of DuckDB's own third party folder. I just couldn't figure out how to include it without making a copy of the file. Happy to adjust that if it is possible to fix.

### CODE REVIEW

File: docs/pages/index.md [hunk 1]
```diff
@@ -34,8 +34,15 @@ The latest version of DuckDB (currently 1.1.2) is required.
...
--- OR create a secret with your Google API access token (boring, see below guide)
+-- OR create a secret with your Google API access token 
+-- (boring, see "Getting a Google API Access Token" below)
 CREATE SECRET (TYPE gsheet, TOKEN '<your_token>');
+
+-- OR create a non-expiring secret with your Google API private key 
+-- (boring, see "Getting a Google API Access Private Key" below)
+-- Note that private key is in the format:
+-- -----BEGIN PRIVATE KEY-----\n ... -----END PRIVATE KEY-----\n
+CREATE SECRET (TYPE gsheet, EMAIL '<service_account_email>', SECRET '<your_private_key>');
```
Comment (archiewood) [inline docs/pages/index.md:45]:
```suggestion
CREATE SECRET (TYPE gsheet, PROVIDER private_key, EMAIL '<service_account_email>', SECRET '<your_private_key>');
```

**Comment by archiewood** [comment]
I may be doing this wrong, but I'm getting the following errors:

with private key pre/suf- fix
CleanShot 2024-11-01 at 21 11 03@2x

removing the pre/suf- fix
CleanShot 2024-11-01 at 21 12 34@2x

File: docs/pages/index.md (hunk 1 above)
Comment (Alex-Monahan) [reply docs/pages/index.md:45]: Yeah, that was a big miss! Thanks!

**Comment by Alex-Monahan** [comment]
Hmmmm.... 

> I may be doing this wrong, but I'm getting the following errors:
> 
> with private key pre/suf- fix CleanShot 2024-11-01 at 21 11 03@2x
> 
> removing the pre/suf- fix CleanShot 2024-11-01 at 21 12 34@2x

Hmmm... Sprry about that! Which version of openssl do you have?

**Comment by archiewood** [comment]
looks like 3.3.0 when i check my vcpkg/info folder

**Comment by archiewood** [comment]
should i be including the full private key string including the begin private key prefix?

i assume yes?

**Comment by Alex-Monahan** [comment]
Yes, both the prefix and suffix should be included. Did you grab it out of that JSON file? That was my approach.

**Pull request closed without merging**